src/Raven.Server/Rachis/Elector.cs

```diff
@@ -37,7 +37,15 @@ public void HandleVoteRequest()
                     using (_engine.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
                     {
                         var rv = _connection.Read<RequestVote>(context);
-
+                        //We are getting a request to vote for our known leader
+                        if (_engine.LeaderTag == rv.Source)
+                        {
+                            _engine.LeaderTag = null;
+                            //If we are followers we want to drop the connection with the leader right away.
+                            //We shouldn't be in any other state since if we are candidate our leaderTag should be null but its safer to verify.
+                            if (_engine.CurrentState == RachisState.Follower)
+                                _engine.SetNewState(RachisState.Follower,null,_engine.CurrentTerm,$"We got a vote request from our leader {rv.Source} so we switch to leaderless state.");
```

Comment:
The problem here is that `SetNewState` will dispose the current state, which is in this case, means us.

Reply:
it will dispose of the follower, we want to do this, i don't follow.

Reply:
Oh, sorry, you are correct. For some reason I thought this is happening on the Follower thread.